code/_helpers/logging.dm

```diff
@@ -37,71 +38,87 @@
 			return
 		if(C.prefs.toggles & CHAT_DEBUGLOGS)
 			C << "DEBUG: [text]"
+	send_gelf_log(short_message=text, long_message="[time_stamp()]: [text]",level=level,category="DEBUG")
 
-/proc/log_game(text)
+/proc/log_game(text,level=5,ckey="",admin_key="",ckey_target="") //FIX target to ckey_target
 	if (config.log_game)
 		game_log("GAME", text)
+	send_gelf_log(short_message=text, long_message="[time_stamp()]: [text]",level=level,category="GAME",additional_data=list("_ckey"=ckey,"_admin_key"=admin_key,"_target"=target))
 
-/proc/log_vote(text)
+/proc/log_vote(text) //Nothing to do
 	if (config.log_vote)
 		game_log("VOTE", text)
+	send_gelf_log(short_message=text, long_message="[time_stamp()]: [text]",level=5,category="VOTE")
 
-/proc/log_access(text)
+/proc/log_access(text,level=5,ckey="") //DONE
 	if (config.log_access)
 		game_log("ACCESS", text)
+	send_gelf_log(short_message=text, long_message="[time_stamp()]: [text]",level=level,category="ACCESS",additional_data=list("_ckey"=ckey))
 
-/proc/log_say(text)
+/proc/log_say(text,level=5,ckey="") //DONE
 	if (config.log_say)
 		game_log("SAY", text)
+	send_gelf_log(short_message=text, long_message="[time_stamp()]: [text]",level=level,category="SAY",additional_data=list("_ckey"=ckey))
 
-/proc/log_ooc(text)
+/proc/log_ooc(text,level=5,ckey="") //DONE
 	if (config.log_ooc)
 		game_log("OOC", text)
+	send_gelf_log(short_message=text, long_message="[time_stamp()]: [text]",level=level,category="OOC",additional_data=list("_ckey"=ckey))
 
-/proc/log_whisper(text)
-	if (config.log_whisper)
+/proc/log_whisper(text,level=5,ckey="") //DONE
+	if (config.log_whisper) 
 		game_log("WHISPER", text)
+	send_gelf_log(short_message=text, long_message="[time_stamp()]: [text]",level=level,category="WHISPER",additional_data=list("_ckey"=ckey))
 
-/proc/log_emote(text)
+/proc/log_emote(text,level=5,ckey="") //DONE
 	if (config.log_emote)
 		game_log("EMOTE", text)
+	send_gelf_log(short_message=text, long_message="[time_stamp()]: [text]",level=level,category="EMOTE",additional_data=list("_ckey"=ckey))
 
-/proc/log_attack(text)
+/proc/log_attack(text,level=5,ckey="",ckey_target="") //DONE
 	if (config.log_attack)
 		game_log("ATTACK", text)
+	send_gelf_log(short_message=text, long_message="[time_stamp()]: [text]",level=5,category="ATTACK",additional_data=list("_ckey"=ckey,"_ckey_target"=ckey_target))
 
-/proc/log_adminsay(text)
+/proc/log_adminsay(text) //Is this even used anywhere ?
 	if (config.log_adminchat)
 		game_log("ADMINSAY", text)
+	send_gelf_log(short_message=text, long_message="[time_stamp()]: [text]",level=5,category="ADMINSAY")
 
-/proc/log_adminwarn(text)
+/proc/log_adminwarn(text) //Is that needed ?

```

Comment:
Adminwarnings should probably be depracted at some point, all of them are sent through log_admin anyways.